owlplug-client/src/main/java/com/owlplug/project/tasks/discovery/studioone/StudioOneAudioMixerPluginCollector.java

```diff
@@ -0,0 +1,172 @@
+      log.error("Error extracting plugins from audio mixer", e);
+    }
+
+    return plugins;
+  }
+
+  private Element findInsertsNode(Document document, XPath xpath) throws XPathExpressionException {
+    // Find the Inserts node by checking x:id attribute directly
+    // Note: x:id is a literal attribute name (colon is part of the name, not a namespace)
+    NodeList allAttributes = (NodeList) xpath.compile("//Attributes")
+            .evaluate(document, XPathConstants.NODESET);
+    for (int i = 0; i < allAttributes.getLength(); i++) {
+      Node node = allAttributes.item(i);
+      if (node instanceof Element elem) {
+        String xidValue = elem.getAttribute("x:id");
+        if ("Inserts".equals(xidValue)) {
+          return elem;
+        }
+      }
+    }
+    return null;
+  }
+
+  private DawPlugin readPluginElement(Element pluginElement) {
+    String pluginName = extractPluginName(pluginElement);
+    if (pluginName == null || pluginName.isEmpty()) {
+      return null;
+    }
+
+    PluginFormat format = extractPluginFormat(pluginElement);
+    if (format == null) {
+      return null;
+    }
+
+    DawPlugin plugin = new DawPlugin();
+    plugin.setName(pluginName);
+    plugin.setFormat(format);
+    return plugin;
+  }
+
+  private String extractPluginName(Element pluginElement) {
+    // For audio effect plugins, prefer classInfo over deviceData
+    // because deviceData may contain instance names (e.g., "Ozone 9", "Ozone 9 (2)")
+    // while classInfo always contains the actual plugin name
+    StudioOneDomUtils.DeviceDataAndGhostData data = StudioOneDomUtils.findDeviceDataAndGhostData(pluginElement);
+    
+    // Try classInfo FIRST (it has the actual plugin name)
+    Element classInfo = StudioOneDomUtils.findClassInfo(data.getGhostData());
+    if (classInfo != null) {
+      String name = classInfo.getAttribute("name");
+      if (name != null && !name.isEmpty()) {
+        return name;
+      }
+    }
+
+    // Fallback to deviceData only if classInfo didn't work
+    if (data.getDeviceData() != null) {
+      String name = data.getDeviceData().getAttribute("name");
+      if (name != null && !name.isEmpty()) {
+        return name;
+      }
+    }
+
+    return null;
+  }
+
+  private PluginFormat extractPluginFormat(Element pluginElement) {
+    StudioOneDomUtils.DeviceDataAndGhostData data = StudioOneDomUtils.findDeviceDataAndGhostData(pluginElement);
+    Element classInfo = StudioOneDomUtils.findClassInfo(data.getGhostData());
+    
+    if (classInfo != null) {
+      String subCategory = classInfo.getAttribute("subCategory");
+      if (subCategory != null && !subCategory.isEmpty()) {
+        // Skip native Studio One plugins
+        if (subCategory.equals("(Native)")) {
+          return null;
+        }
+
+        // Determine format from subCategory (format is like "VST2/Mastering" or "VST3/Mastering")
+        if (subCategory.startsWith("VST2") || subCategory.contains("VST2")) {
```

Comment:
It works, but the logic is redundant and can be improved.

You're checking whether subCategory contains "VST2".
Because `.startsWith("VST2")` is already covered by `.contains("VST2")`, the startsWith part isn’t needed.